internal/engine/upper_test.go

```diff
@@ -2728,8 +2728,9 @@ func newTestFixture(t *testing.T) *testFixture {
 	reaper := build.NewImageReaper(dockerClient)
 
 	kCli := k8s.NewFakeK8sClient()
-	pw := NewPodWatcher(kCli)
-	sw := NewServiceWatcher(kCli, "")
+	of := k8s.ProvideOwnerFetcher(kCli)
```

Comment:
i feel like we eventually need a pattern for doing dependency injection for test fixtures. they're pretty annoying to set up right now